### PR TITLE
Adding Java-Specific typekinds #3463

### LINKED ISSUE #3463: Support more Java-specific APIView 'TypeKinds'
It would be nice to have support for additional TypeKinds in APIView that relate to Java. I would like the following: 

* ANNOTATION
* JAVA_MODULE
* MAVEN
* GRADLE

This requires changes to the Java APIView tooling to include these as appropriate, as well as the APIViewWeb tooling to make appropriate use of these type kinds (most notably, to have more unique icons in the left navigation bar).

### PR DESCRIPTION
**Pull request by Baas-hub**
Adding icons and the custom scss data to change sources from similar to this:
image
To be more like:
image

This fixes #3463 and relies on #3464 (PR: #3978) for the custom scss/css files before being merged.

### CODE REVIEW

File: src/dotnet/APIView/APIViewWeb/Client/css/kotlin.scss [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+.icon-spring {
+    background: url(/icons/java/common/spring.svg) center left no-repeat !important;
+}
+
+.icon-unknown {
+    background: url(/icons/java/common/unknown.svg) center left no-repeat !important;
+}
+
+.icon-xamarinAndroid {
```
Comment (JonathanGiles) [inline src/dotnet/APIView/APIViewWeb/Client/css/kotlin.scss:25]: Let's change every reference of `xamarinAndroid` to just `android`

File: src/java/apiview-java-processor/src/main/java/com/azure/tools/apiview/processor/model/TypeKind.java [hunk 2]
```diff
@@ -8,6 +8,10 @@ public enum TypeKind {
     CLASS("class"),
     INTERFACE("interface"),
     ENUM("enum"),
+    ANNOTATION("annotation"),
+    JAVA_MODULE("java_module"),
```
Comment (JonathanGiles) [inline src/java/apiview-java-processor/src/main/java/com/azure/tools/apiview/processor/model/TypeKind.java:12]: I would just change this to be 'MODULE' with "module", and rename the css selectors appropriately.

**Review by JonathanGiles — APPROVED** [review]
Minor nits, but otherwise looks great!

File: src/java/apiview-java-processor/src/main/java/com/azure/tools/apiview/processor/analysers/JavaASTAnalyser.java [hunk 3]
```diff
@@ -1322,7 +1322,7 @@ private class ScanForClassTypeVisitor extends VoidVisitorAdapter<Map<String, Str
         @Override
         public void visit(CompilationUnit compilationUnit, Map<String, String> arg) {
             compilationUnit.getModule().ifPresent(moduleDeclaration ->
-                apiListing.addChildItem(new ChildItem(MODULE_INFO_KEY, MODULE_INFO_KEY, TypeKind.CLASS)));
+                apiListing.addChildItem(new ChildItem(MODULE_INFO_KEY, MODULE_INFO_KEY, TypeKind.JAVA_MODULE)));
```
Comment (JonathanGiles) [inline src/java/apiview-java-processor/src/main/java/com/azure/tools/apiview/processor/analysers/JavaASTAnalyser.java:1325]: Fix this to be MODULE now :)

**Comment by JonathanGiles** [comment]
Oh - one other comment - the screenshot above doesn't seem to show the new icon for 'package' - it is still showing the old icon. If you can just double check all icons are getting replaced, that would  be great!

**Review by praveenkuttappan — APPROVED** [review]

**Pull request merged**